### PR TITLE
Fix kerberos under yarn: add delegation token search

### PR DESCRIPTION
**Pull request by alexr-bq**
### Summary

Check for existing delegation token before requesting new one. This is required for YARN support.

### Related Issue

VER-77541

### Additional Reviewers

@jonathanl-bq 
@ravjotbrar

### CODE REVIEW

File: connector/src/main/scala/com/vertica/spark/datasource/fs/FileStoreLayerInterface.scala [hunk 1]
```diff
@@ -420,25 +426,41 @@ class HadoopFileStoreLayer(fileStoreConfig : FileStoreConfig, schema: Option[Str
...
+    var hdfsToken: Option[String] = None
+    val ugiUser = UserGroupInformation.getLoginUser
+    if(ugiUser != null) {
+      logger.debug("Got UGI user.")
+      val existingTokens = ugiUser.getCredentials.getAllTokens
+      val itr = existingTokens.iterator
+      while (itr.hasNext) {
+        val token = itr.next();
+        logger.debug("Existing token kind: " + token.getKind.toString)
+        if (token.getKind.equals(new Text("HDFS_DELEGATION_TOKEN"))) {
```
Comment (NerdLogic) [inline connector/src/main/scala/com/vertica/spark/datasource/fs/FileStoreLayerInterface.scala:440]: Can this be done with `ugiUser.getCredentials.getToken(new Text("HDFS_DELEGATION_TOKEN")`?

File: connector/src/main/scala/com/vertica/spark/datasource/fs/FileStoreLayerInterface.scala [hunk 2]
```diff
@@ -420,25 +426,41 @@ class HadoopFileStoreLayer(fileStoreConfig : FileStoreConfig, schema: Option[Str
 
   // scalastyle:off
   override def getImpersonationToken(user: String) : ConnectorResult[String] = {
-    this.useFileSystem(fileStoreConfig.address, (fs, _) => {
-      var hdfsToken = ""
 
+    // First, see if there is already a delegation token (this is the case when run under YARN)
+    var hdfsToken: Option[String] = None
+    val ugiUser = UserGroupInformation.getLoginUser
```
Comment (jonathanl-bq) [inline connector/src/main/scala/com/vertica/spark/datasource/fs/FileStoreLayerInterface.scala:432]: Option(UserGroupInformation.getLoginUser)

**Review by jonathanl-bq — CHANGES_REQUESTED** [review]

File: connector/src/main/scala/com/vertica/spark/datasource/fs/FileStoreLayerInterface.scala (hunk 1 above)
Comment (alexr-bq) [reply connector/src/main/scala/com/vertica/spark/datasource/fs/FileStoreLayerInterface.scala:440]: Tried this, it failed.

File: connector/src/main/scala/com/vertica/spark/datasource/fs/FileStoreLayerInterface.scala (hunk 2 above)
Comment (alexr-bq) [reply connector/src/main/scala/com/vertica/spark/datasource/fs/FileStoreLayerInterface.scala]: done

**Review by jonathanl-bq — APPROVED** [review]

**Pull request merged**